### ISSUE TITLE
Bug in makespec.Path's __repr__ method on windows.

### ISSUE THREAD

**Issue by Legorooj** [pyinstaller/pyinstaller#4820]
@htgoebel I'm stuck on this one, and I'm generally quite good at path manipulation.

```
______________________________________________________________________________________________ test_gevent[onefile] ______________________________________________________________________________________________

pyi_builder = <PyInstaller.utils.conftest.AppBuilder object at 0x09C58F70>

    @importorskip('gevent')
    def test_gevent(pyi_builder):
        pyi_builder.test_source(
            """
            import gevent
            gevent.spawn(lambda: x)
>           """)

D:\PyInstaller\tests\functional\test_libraries.py:93:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
D:\PyInstaller\PyInstaller\utils\conftest.py:444: in _test_building
    pyi_main.run(pyi_args, PYI_CONFIG)
D:\PyInstaller\PyInstaller\__main__.py:112: in run
    spec_file = run_makespec(**vars(args))
D:\PyInstaller\PyInstaller\__main__.py:58: in run_makespec
    spec_file = PyInstaller.building.makespec.main(filenames, **opts)
D:\PyInstaller\PyInstaller\building\makespec.py:463: in main
    specfile.write(onefiletmplt % d)
D:\PyInstaller\PyInstaller\building\makespec.py:106: in __repr__
    self.variable_prefix, self.filename_suffix = make_variable_path(self.path)
D:\PyInstaller\PyInstaller\building\makespec.py:89: in make_variable_path
    if os.path.commonpath([filename, from_path]) == from_path:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

paths = ('C:\\Users\\legor\\AppData\\Local\\Temp\\pytest-of-legor\\pytest-24\\test_gevent_onefile_0\\test_source.py', 'D:\\PyInstaller')

    def commonpath(paths):
        """Given a sequence of path names, returns the longest common sub-path."""

        if not paths:
            raise ValueError('commonpath() arg is an empty sequence')

        paths = tuple(map(os.fspath, paths))
        if isinstance(paths[0], bytes):
            sep = b'\\'
            altsep = b'/'
            curdir = b'.'
        else:
            sep = '\\'
            altsep = '/'
            curdir = '.'

        try:
            drivesplits = [splitdrive(p.replace(altsep, sep).lower()) for p in paths]
            split_paths = [p.split(sep) for d, p in drivesplits]

            try:
                isabs, = set(p[:1] == sep for d, p in drivesplits)
            except ValueError:
                raise ValueError("Can't mix absolute and relative paths") from None

            # Check that all drive letters or UNC paths match. The check is made only
            # now otherwise type errors for mixing strings and bytes would not be
            # caught.
            if len(set(d for d, p in drivesplits)) != 1:
>               raise ValueError("Paths don't have the same drive")
E               ValueError: Paths don't have the same drive

..\..\..\..\Programs\Python\Python37-32\lib\ntpath.py:621: ValueError
```

It appears that running the tests - well `makespec` is generating the error - on a different drive doesn't work. I'm not sure there's a way around this.

**Comment by Legorooj** [comment]
Figured out the rough cause of the problem. We're trying to use relative paths across seperate drives. I need to look into this more.

**Comment by bjones1** [comment]
This was introduced in 042d21cec66308f39ef530bf0e6dbd5de3fa442c. I'm working on a fix.

**Comment by Legorooj** [comment]
@bjones1 you got there before I did... I'd only just figured out the problem. I hadn't even gotten to a fix.

**Comment by Legorooj** [comment]
Odd. https://travis-ci.org/github/bjones1/pyinstaller/jobs/683086293#L924 is failing on an exitcode of 0.

**Comment by Legorooj** [comment]
@bjones1 can confirm fix in 886d932 works. Open a PR and assign me when we've got these macOS builds working.

**Comment by bjones1** [comment]
I'm waiting for tests on my fix to finish, then I'll push. This should land tomorrow.
